### ISSUE TITLE
Row Group

### ISSUE THREAD

**Issue by food-data-sci** [glin/reactable#112]
Is there a way to group rows in reactable and create headers, like the rowGroup functionality you can get in DT?

![image](https://user-images.githubusercontent.com/55811656/103657796-40f32d00-4f62-11eb-8bba-c0ee7e4637eb.png)
screenshot from https://rstudio.github.io/DT/extensions.html

I know you can group them and collapse the whole group but the indentation you get after the group doesn't look great with multiple levels of grouping. + I want the 'headers' visible without a need to expand rows. 

Perhaps there's a way to get from collapsible groups to a headed table, like the one in the example I pasted?

**Comment by glin** [comment]
No, this isn't possible today. The `reactable(defaultExpanded = TRUE)` option could remove the need to expand rows, but there's no way to change the row grouping behavior. However, I've had the same thought about indentation before, and always wished there was a way to create a flatter nested/tree table structure. A feature like that may come in the future, but I can't say when right now.

**Comment by algo-se** [comment]
A _"a flatter nested/tree table structure"_ sounds awesome tbh 😄 I think that creating a tree structure in a table is much more powerful than using shinytree, etc.

[jsTreeR](https://github.com/stla/jsTreeR) is a package that kind of have this feature (see the jsTree grid) but nothing close to reactable.
